src/player.cpp

```diff
@@ -7841,9 +7841,15 @@ static const std::array<matype_id, 4> bio_cqb_styles {{
 
 class ma_style_callback : public uimenu_callback
 {
+private:
+    const input_context ctxt;
```

Comment:
Why a copy and not just a reference?

Reply:
Mainly I wanted to avoid introducing weak pointers as class members without a good reason. They make reasoning about semantics a bit more complex. input_context looked like it was just a bunch of strings, so the copy didn't seem overly expensive. The action_map member (I think this was in the item_action callback class) similarly was a collection of strings and copied iirc.

If I get time later today or tomorrow, I'll change the callback interface to pass the input_context by ref to const so the class member won't be necessary. I just started playing again yesterday, and I already put more time to deving than playing. My first character is barely out of the shelter. hah

Reply:
Most recent commits in pr have removed this input_context member.